### PR TITLE
Improve performance of source generator

### PR DESCRIPTION
**Pull request by manodasanW**
- Changed the ordering of the properties during the combine operations to allow us to know earlier whether authoring support is enabled.  This allows us to reduce how often we call `GetVtableAttributesToAddOnLookupTable` and thereby reducing calls to `GetSymbolInfo`.  This should reduce the number of calls by half and thereby improve performance.

### CODE REVIEW

File: src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs [hunk 1]
```diff
@@ -19,8 +19,28 @@ public class WinRTAotSourceGenerator : IIncrementalGenerator
...
+                    // check for such as component support where the generator introduces the reference and embedded support.
+                    // We will allow for an override for this by detecting whether CsWinRTAotWarningLevel is set to 2 which
+                    // indicates the component cares about optimization for built-in interfaces.
+                    (value.properties.isCsWinRTAotOptimizerEnabled && 
+                     (value.properties.isCsWinRTComponent ||
+                      value.properties.isCsWinRTAotOptimizerEnabledForBuiltInInterfaces ||
+                      value.properties.isCsWinRTEmbeddedSupport ||
+                      GeneratorHelper.HasWinRTRuntimeReference(value.compilation, ct)), 
+                     value.properties.isCsWinRTComponent,
+                     value.properties.Item3));
```
Comment (Sergio0694) [inline src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs:43]: Why are we returning the token..?

File: src/Authoring/WinRT.SourceGenerator/Helper.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@
 using System.IO;
 using System.Linq;
 using System.Text.RegularExpressions;
+using System.Threading;
```
[the remark this answers is not in the document]
Comment (manodasanW) [reply src/Authoring/WinRT.SourceGenerator/Helper.cs:14]: This is from CancellationToken.

File: src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs (hunk 1 above)
Comment (manodasanW) [reply src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs:43]: which token?  Item3 is another bool which I am using to reconstruct the properties that is used later.

Comment (Sergio0694) [reply src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs:43]: Oh, sorry, misread. Too many tuple items ahah

File: src/Authoring/WinRT.SourceGenerator/Helper.cs (hunk 2 above)
Comment (Sergio0694) [inline src/Authoring/WinRT.SourceGenerator/Helper.cs:14]: Leftover now? It seems you removed the cancellation token?

**Review by Sergio0694 — APPROVED** [review]
LGTM!

**Pull request merged**